### PR TITLE
Sieve of atkins in C++

### PR DESCRIPTION
**Pull request by ShubhamGupta577**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

**Adding the code for Sieve_Of_Atkins in C++**
Sieve of Atkin is a modern algorithm for finding all prime numbers up to a specified integer. Compared with the ancient Sieve of Eratosthenes, which marks off multiples of primes, the Sieve of Atkin does some preliminary work and then marks off multiples of squares of primes, thus achieving a better theoretical asymptotic complexity.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Issue Resolved - (#2665 )

### CODE REVIEW

**Comment by ShubhamGupta577** [comment]
@bhav09 Please review this.

File: C-Plus-Plus/math/Sieve_Of_Atkins.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/*
+	Sieve of Atkin is a modern algorithm for finding all prime numbers up to a
+	specified integer. Compared with the ancient Sieve of Eratosthenes, which
+	marks off multiples of primes, the Sieve of Atkin does some preliminary work
+	and then marks off multiples of squares of primes, thus achieving a better 
+	theoretical asymptotic complexity.
+
+	Sample input :
```
Comment (bhav09) [inline C-Plus-Plus/math/Sieve_Of_Atkins.cpp]: Can you mention I/O at the end of the code. Rest everything is picture perfect!

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed.

**Comment by ShubhamGupta577** [comment]
@bhav09 Requested changes have been made.

File: C-Plus-Plus/math/Sieve_Of_Atkins.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+
+/*
+	Sample input :
+ 	43
+ 	Sample output:
+ 	2 3 5 7 11 13 17 19 23 29 31 37 41
+	Time Complexity -O(n/(log(logn)))
+	Space Complexity-O(n^0.5)
+*/
```
Comment (bhav09) [inline C-Plus-Plus/math/Sieve_Of_Atkins.cpp:92]: Great work!
Can you add a new line at the end here. Rest everything is cool

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed.

**Comment by ShubhamGupta577** [comment]
@bhav09 Requested Change has been done. Please review it

**Comment by bhav09** [comment]
Great work! Keep it up :)

**Pull request merged**